### ISSUE TITLE
Mouse locked in single place when switching out of VR mode; Requires quitting down to fix

### ISSUE THREAD

**Issue by FlameSoulis** [Yellow-Dog-Man/Resonite-Issues#670]
### Describe the bug?

Sometimes, when switching out of VR mode, the mouse will get locked to a specific place and not move, including when the app is focused. As a result, it's impossible to access any menu, any other application, and ultimately prevents the operation of an entire computer system! The ONLY solution at this point is to ensure Resonite is focused and use Alt+F4 to force a shutdown of the app.

### To Reproduce

1. Launch Resonite while in VR
2. Take headset off
3. Randomly notice the mouse is locked, no matter what

### Expected behavior

1. Launch Resonite while in VR
2. Take headset off
3. Be able to mouse the mouse normally

### Resonite Version Number

2023.11.13.1343

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop, Index

### Log Files

[DEMONDOGGO-PC - 2023.11.13.1343 - 2023-11-13 18_24_23.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13343239/DEMONDOGGO-PC.-.2023.11.13.1343.-.2023-11-13.18_24_23.log)

### Additional Context

The issue was noticed on earlier versions (3 builds ago?).

Additionally, unlike #242, the mouse isn't in the center of the display: rather, it's always been on the lower left, offset slightly from the center. Additionally, that bug unlocks the mouse when unfocused, whereas this doesn't.

I'll do some more testing on the possible cause, but I'm wondering if something is now buggy with the Desktop viewer.

### Reporters

U-Flame-Soulis (flamesoulis)

**Comment by Nytra** [comment]
So I think I've had this happen to me. Whenever it happens if you press Ctrl+Alt+Delete and then click the Task Manager button you will be able to move the mouse again.

**Comment by FlameSoulis** [comment]
> So I think I've had this happen to me. Whenever it happens if you press Ctrl+Alt+Delete and then click the Task Manager button you will be able to move the mouse again.

I think this only furthers the point the desktop viewer is bugged: Resonite usually doesn't run as an admin, so it cannot control the mouse if a program with higher privileges is focused.

**Comment by DanteTucker** [comment]
Either Esc or Alt-Tab should also free the mouse.

**Comment by TisFoolish** [comment]
I've had it happen just starting out in Desktop mode, so it's not caused by the switch.  Also, alt tabbing does not free the mouse, nor does clicking on a window under the mouse if you did alt tab out.

**Comment by FlameSoulis** [comment]
> Either Esc or Alt-Tab should also free the mouse.

Nope. This does not fix the issue at all. Even the Start/Super key doesn't work. You NEED to destroy Resonite's ability to move the mouse.

**Comment by FlameSoulis** [comment]
Updated the ticket: It's the Desktop Viewer and the auto-swap feature introduced to Resonite causing the issue.

**Comment by ProtogenDelta** [comment]
Flagging to follow post-merge, but this is likely caused by whatever logic is in charge of detecting mouse movement in desktop mode, since it would likely be re-centering or otherwise locking the mouse to ensure it doesn't run off screen and stop providing events. Obviously without being able to see the code, my ability to comment on the specifics is limited, however this should easily be addressed with a proper focus check when switching modes.
